tests/freertos/semaphore_test.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2019 Julian Holzwarth <julian.holzwarth@fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "freertos/semphr.h"
+#include "freertos/queue.h"
+#include "freertos/FreeRTOS.h"
+
+#include "semaphore_test.h"
+
+/*
+ * @brief amount of times the tested function will be called
+ */
+#define TEST_SEMAPHORE_FOR_COUNTER 20
+
+/**
+ * @brief   tests the take function for a freertos mutex semaphore
+ *
+ * @return pdPASS when the test is passed, pdFail otherwise
+ */
+int semaphore_test_mutex_take(void)
+{
+    SemaphoreHandle_t testing_semaphore = xSemaphoreCreateMutex();
+    if (testing_semaphore == NULL) {
+        puts("test failed: mutex semaphore not created");
+    }
+    /*
+    Freertos Documentation:
+    "pdPASS: Returned only if the call to xSemaphoreTake()
+    was successful in obtaining the semaphore."
+    */
+    /* first call should be successful */
+    if (xSemaphoreTake(testing_semaphore,0) == pdFAIL) {
+        puts("error in Take");
+        return pdFAIL;
+    }
+    /* after the fist call every call should fail*/
+    for(size_t i = 0; i < TEST_SEMAPHORE_FOR_COUNTER; i++)
```

Comment:
Why 20 times? One further try to lock the mutex should be enough to test the functionality that is tested here.